sync/src/chain.rs

```diff
@@ -437,6 +448,10 @@ impl ChainSync {
 		trace!(target: "sync", "{} -> NewBlock ({})", peer_id, h);
 		let header: BlockHeader = try!(header_rlp.as_val());
  		let mut unknown = false;
+		{
+			let peer = self.peers.get_mut(&peer_id).expect("ChainSync: unknown peer");
+			peer.latest = header.hash();
```

Comment:
Update peer's latest hash regardless if the block is known or not, this prevents sending out blocks peer already knows